extensions-contrib/opencensus-extensions/src/main/java/org/apache/druid/data/input/opencensus/protobuf/OpenCensusProtobufInputFormat.java

```diff
@@ -59,6 +69,35 @@ public boolean isSplittable()
   @Override
   public InputEntityReader createReader(InputRowSchema inputRowSchema, InputEntity source, File temporaryDirectory)
   {
+    try {
+      KafkaRecordEntity kafkaInputEntity = (KafkaRecordEntity) source;
+      Header versionHeader = kafkaInputEntity.getRecord().headers().lastHeader(VERSION_HEADER_KEY);
+      int version = 0;
+      if (versionHeader != null) {
+        version = ByteBuffer.wrap(versionHeader.value()).order(ByteOrder.LITTLE_ENDIAN).getInt();
+      }
+
+      if (version == 1) {
+        return new OpenTelemetryMetricsProtobufReader(
+          inputRowSchema.getDimensionsSpec(),
+          (ByteEntity) source,
```

Comment:
no need to cast here, you can pass kafkaInputEntity directly since KafkaRecordEntity is a subclass of ByteEntity

Reply:
I guess this was lazy copy paste. You are right, updated